blocks/ticket/ticket.ts

```diff
@@ -150,10 +169,22 @@ export class Ticket {
                         const mapping =
                           ticketingWorkflowVarMapping[v.fieldType];
 
-                        const filteredVars = allWorkflowVars.filter((v) =>
-                          mapping ? mapping.includes(v.type) : true
+                        let filteredVars = allWorkflowVars.filter(
+                          (workflowVar) =>
+                            !mapping ||
+                            mapping?.includes(workflowVar.type) ||
+                            // for single select field, we allow COMP variable
+                            (v.fieldType === "SEL" &&
+                              /^COMP\d+/.test(workflowVar.label))
                         );
 
+                        // For requestor field, cannot be none
+                        if (v.name !== "Requestor") {
+                          filteredVars = [
+                            { value: "", label: "--None--", type: "" },
+                          ].concat(filteredVars);
+                        }
```

Comment:
Special requirement from product, in key value mapping all need to have '--None--' deselector except requestor field

Reply:
probably introduce some new toggle to disable or enable none. allowUnselect or something. what do you think?

Reply:
Update in SelectInput component in client to support this

Reply:
sounds good. For this "Requestor" field, it is not defined in cbk, but is retrieved from ticket endpoint so this part that hard code check for "Requestor" has to stay the same, or add a new toggle in FE and database. Is that alright?

Reply:
yes new toggle in fe. maybe can reuse isReadOnly to toggle for this specific requestor 

{"isDefault": true, "isReadOnly": true}

Reply:
Ahh right, I will use isReadOnly, thanks!